### PR TITLE
[IOTDB-700]Feature/iotdb 700 add openid

### PR DESCRIPTION
**Pull request by JulianFeinauer**
As discussed with @jixuan1989 in Slack he will have a look on this Branch and we will discuss how to integrate multiple Auth Providers in the config.

### CODE REVIEW

File: server/src/main/java/org/apache/iotdb/db/auth/authorizer/OpenIdAuthorizer.java [hunk 1]
```diff
@@ -0,0 +1,248 @@
...
+    OpenIdAuthorizer(String providerUrl) throws AuthException, URISyntaxException, ParseException, IOException {
+        this(getJWKfromProvider(providerUrl));
+    }
+
+    private static JSONObject getJWKfromProvider(String providerUrl) throws URISyntaxException, IOException, ParseException, AuthException {
+        if (providerUrl == null) {
+            throw new IllegalArgumentException("OpenID Connect Provider URI must be given!");
+        }
+
+        //
```
Comment (xingtanzjr) [inline server/src/main/java/org/apache/iotdb/db/auth/authorizer/OpenIdAuthorizer.java:75]: Let's remove empty comment here : )

Comment (jixuan1989) [reply server/src/main/java/org/apache/iotdb/db/auth/authorizer/OpenIdAuthorizer.java:75]: welcome back @xingtanzjr

File: server/src/main/java/org/apache/iotdb/db/auth/authorizer/LocalFileAuthorizer.java [hunk 2]
```diff
@@ -19,44 +19,32 @@
 package org.apache.iotdb.db.auth.authorizer;
 
 import java.io.File;
+import java.io.IOException;
+import java.net.URISyntaxException;
+
+import com.nimbusds.oauth2.sdk.ParseException;
```
Comment (jixuan1989) [inline server/src/main/java/org/apache/iotdb/db/auth/authorizer/LocalFileAuthorizer.java:25]: the unused "import" should be removed..

File: server/src/test/java/org/apache/iotdb/db/auth/authorizer/OpenIdAuthorizerTest.java [hunk 3]
```diff
@@ -0,0 +1,64 @@
...
+
+        OpenIdAuthorizer authorizer = new OpenIdAuthorizer(JSONObjectUtils.parse(OPEN_ID_PUBLIC_JWK));
+        boolean login = authorizer.login(jwt, null);
```
Comment (jixuan1989) [inline server/src/test/java/org/apache/iotdb/db/auth/authorizer/OpenIdAuthorizerTest.java:28]: has to consider how to write the Test. 
Either use a Mock or use a stable OpenID server.

**Comment by JulianFeinauer** [comment]
@jixuan1989 I added some english users documentation and fixed the sonar issues (the last I think is fine, better log more than less). So I am open for any further feedback!

**Comment by jixuan1989** [comment]
Hi @JulianFeinauer , pls pay attention to then the Travis error  (using WinOS, some resource is not released when the JWT connection fails. If we do not fix the issue, then all Travis after merging this PR will fail...)

**Comment by jixuan1989** [comment]
As for the sonar issue about the log, I think following Sonar tips is fine.

As you have already thrown the exception to the caller. The handler who receives the exception should handle the exception or print the exception. 

But in this case, the exception log may be printed twice.

image

**Comment by JulianFeinauer** [comment]
@jixuan1989 I fixed everything. I hope you are happy with me now? :)

**Comment by jixuan1989** [comment]
Hi @JulianFeinauer , Look good to me. 
I will add the Chinese user guide later.
How about add this feature into v0.10.

**Pull request merged**